### ISSUE TITLE
handle wrongly passed values

### ISSUE THREAD

**Issue by sakulstra** [bgd-labs/aave-debt-swap#16]
The contract currently leaves 0 leftovers when used correctly.

When you pass a `debtAmount > your actual debt`, but not `type(uint256).max` there will be leftovers on the contract.

Probably this check should be enhanced to account for misusage:
https://github.com/bgd-labs/aave-debt-swap/blob/main/src/contracts/ParaSwapDebtSwapAdapter.sol#L80-L85

e.g. revert when `amount > debt && amount != max`

**Comment by sakulstra** [comment]
Perhaps instead of reverting one could also just min/max this

```diff
-if (debtSwapParams.debtRepayAmount == type(uint256).max) {
-      (address vToken, address sToken, ) = _getReserveData(debtSwapParams.debtAsset);
-      debtSwapParams.debtRepayAmount = debtSwapParams.debtRateMode == 2
-        ? IERC20WithPermit(vToken).balanceOf(msg.sender)
-        : IERC20WithPermit(sToken).balanceOf(msg.sender);
-    }
+ uint256 maxDebtRepayAmount = debtSwapParams.debtRateMode == 2
+   ? IERC20WithPermit(vToken).balanceOf(msg.sender)
+   : IERC20WithPermit(sToken).balanceOf(msg.sender);
+ if(debtSwapParams.debtRepayAmount > maxDebtRepayAmount) 
+ debtSwapParams.debtRepayAmount = maxDebtRepayAmount;
```

**Comment by Zer0dot** [comment]
> Perhaps instead of reverting one could also just min/max this
> 
> ```diff
> -if (debtSwapParams.debtRepayAmount == type(uint256).max) {
> -      (address vToken, address sToken, ) = _getReserveData(debtSwapParams.debtAsset);
> -      debtSwapParams.debtRepayAmount = debtSwapParams.debtRateMode == 2
> -        ? IERC20WithPermit(vToken).balanceOf(msg.sender)
> -        : IERC20WithPermit(sToken).balanceOf(msg.sender);
> -    }
> + uint256 maxDebtRepayAmount = debtSwapParams.debtRateMode == 2
> +   ? IERC20WithPermit(vToken).balanceOf(msg.sender)
> +   : IERC20WithPermit(sToken).balanceOf(msg.sender);
> + if(debtSwapParams.debtRepayAmount > maxDebtRepayAmount) 
> + debtSwapParams.debtRepayAmount = maxDebtRepayAmount;
> ```

I think this is the best solution, if `repayAmount > maxDebtToRepay` default it to that max. 

Another option would be simply to send back the remaining swapped amount, effectively allowing this to act as a 1tx borrow & swap, without even needing any debt, but that is a different thing and I'm thinking we stick to the solution you outlined.

**Comment by Zer0dot** [comment]
Addressed this in #17 at commit [69603c1](https://github.com/bgd-labs/aave-debt-swap/pull/17/commits/69603c113e8707e4d5897ecb8e652e5c7f4e025b)
